python/pyspark/sql/types.py

```diff
@@ -1119,8 +1120,8 @@ def Dict(d):
     class Row(tuple):
 
         """ Row in DataFrame """
-        __DATATYPE__ = dataType
-        __FIELDS__ = tuple(f.name for f in dataType.fields)
+        __datatype = dataType
```

Comment:
Due to name-mangling, this field will no longer be accessible outside of the `Row` class itself, but I don't think that's a problem based on how we use it: it looks like the only place where we read the old `__DATATYPE__` field was in `__reduce__`.